libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/extensions/RoutesExtraEx.kt

```diff
@@ -0,0 +1,19 @@
+package com.mapbox.navigation.core.internal.extensions
+
+import com.mapbox.navigation.core.directions.session.RoutesExtra.ROUTES_UPDATE_REASON_ALTERNATIVE
+import com.mapbox.navigation.core.directions.session.RoutesExtra.ROUTES_UPDATE_REASON_CLEAN_UP
+import com.mapbox.navigation.core.directions.session.RoutesExtra.ROUTES_UPDATE_REASON_NEW
+import com.mapbox.navigation.core.directions.session.RoutesExtra.ROUTES_UPDATE_REASON_REFRESH
+import com.mapbox.navigation.core.directions.session.RoutesExtra.ROUTES_UPDATE_REASON_REROUTE
+import com.mapbox.navigator.SetRoutesReason
+
+fun String.toSetRoutesReason(): SetRoutesReason {
+    return when (this) {
+        ROUTES_UPDATE_REASON_CLEAN_UP -> SetRoutesReason.CLEAN_UP
+        ROUTES_UPDATE_REASON_NEW -> SetRoutesReason.NEW_ROUTE
+        ROUTES_UPDATE_REASON_ALTERNATIVE -> SetRoutesReason.ALTERNATIVE
+        ROUTES_UPDATE_REASON_REROUTE -> SetRoutesReason.REROUTE
+        ROUTES_UPDATE_REASON_REFRESH -> SetRoutesReason.NEW_ROUTE
```

Comment:
should it be `NEW_ROUTE `?

Reply:
We should never end up in this code path.`ROUTES_UPDATE_REASON_REFRESH` doesn't set routes to navigator, it uses a separate function to update the routes.

Reply:
Perhaps we can pass the reason string all the way down to native navigator and only map then, when we definitely won't run into refresh.

Reply:
@LukasPaczos if we do so, we need to change `RoutesExtra` module, to make it visible from `lib-navigator`, it will break imports (if we want to use conts from `RoutesExtra`)
also we can map like 
`"ROUTES_UPDATE_REASON_NEW" -> SetRoutesReason.NEW_ROUTE`
but I think we should avoid it

Reply:
```diff
diff --git a/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt b/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt
index d6f08d49a4..e2004fb166 100644
--- a/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt
+++ b/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt
@@ -18,7 +18,7 @@ import com.mapbox.navigation.core.BasicSetRoutesInfo
 import com.mapbox.navigation.core.SetAlternativeRoutesInfo
 import com.mapbox.navigation.core.SetRefreshedRoutesInfo
 import com.mapbox.navigation.core.SetRoutesInfo
-import com.mapbox.navigation.core.internal.extensions.toSetRoutesReason
+import com.mapbox.navigation.core.directions.session.RoutesExtra
 import com.mapbox.navigation.core.navigator.getCurrentBannerInstructions
 import com.mapbox.navigation.core.navigator.getLocationMatcherResult
 import com.mapbox.navigation.core.navigator.getRouteProgressFrom
@@ -43,6 +43,7 @@ import com.mapbox.navigator.NavigationStatusOrigin
 import com.mapbox.navigator.NavigatorObserver
 import com.mapbox.navigator.RouteAlternative
 import com.mapbox.navigator.RouteState
+import com.mapbox.navigator.SetRoutesReason
 import kotlinx.coroutines.CoroutineStart
 import kotlinx.coroutines.Job
 import kotlinx.coroutines.cancelChildren
@@ -150,7 +151,7 @@ internal class MapboxTripSession(
 
     private suspend fun setRouteToNativeNavigator(
         routes: List<NavigationRoute>,
-        setRoutesInfo: SetRoutesInfo
+        setRoutesInfo: BasicSetRoutesInfo
     ): NativeSetRouteResult {
         logD(
             "native routes update (route IDs: ${routes.map { it.id }}) - starting",
@@ -160,7 +161,7 @@ internal class MapboxTripSession(
         return navigator.setRoutes(
             newPrimaryRoute,
             setRoutesInfo.legIndex,
-            setRoutesInfo.reason.toSetRoutesReason(),
+            setRoutesInfo.toSetRoutesReason(),
             routes.drop(1)
         ).onValue {
             updateLegIndexJob?.cancel()
@@ -708,3 +709,12 @@ internal class MapboxTripSession(
         }
     }
 }
+
+private fun BasicSetRoutesInfo.toSetRoutesReason(): SetRoutesReason {
+    return when (this.reason) {
+        RoutesExtra.ROUTES_UPDATE_REASON_CLEAN_UP -> SetRoutesReason.CLEAN_UP
+        RoutesExtra.ROUTES_UPDATE_REASON_NEW -> SetRoutesReason.NEW_ROUTE
+        RoutesExtra.ROUTES_UPDATE_REASON_REROUTE -> SetRoutesReason.REROUTE
+        else -> throw IllegalArgumentException("Invalid routes update reason: $this")
+    }
+}
```

Just as an idea but ideally we should split up the `BasicSetRoutesInfo` into each respective `new`, `cleanup`, `reroute` sub-types.

Reply:
> ideally we should split up the BasicSetRoutesInfo into each respective new, cleanup, reroute sub-types.

let's do it separately

Reply:
NAVAND-891